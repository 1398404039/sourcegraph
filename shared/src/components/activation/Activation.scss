.activation-status-dropdown-button {
    &--hidden {
        display: none;
    }

    &__progress-bar-container {
        display: inline-block;
        width: 1rem;
        margin-left: 0.4rem;

        .CircularProgressbar-trail {
            stroke: $dropdown-border-color;
        }
        .CircularProgressbar-path {
            stroke: $oc-green-4;
        }
    }
}

.activation-status-dropdown-header {
    padding: 0.25rem 0.5rem;
}

.activation-list {
    margin-top: 0.5rem;
}

.activation-list__item {
    padding-left: 0.25rem;
    padding-right: 0.25rem;
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;

    border-bottom: 1px solid $dropdown-border-color;

    @include hover-focus {
        color: $dropdown-link-hover-color;
        text-decoration: none;
        @include gradient-bg($dropdown-link-hover-bg);
    }
    &:last-child {
        border-bottom: none;
    }
}

.activation-item {
    cursor: pointer;
    padding-left: 0.25rem;
    padding-right: 0.25rem;
    padding-top: 0.25rem;
    padding-bottom: 0.25rem;

    &__checkbox {
        &--done {
            color: $oc-green-6;
        }
        &--todo {
            color: $dropdown-border-color;
        }
    }
}

.first-use-button {
    display: block;
    position: relative;
}

.first-use-button::before,
.first-use-button::after {
    position: absolute;
    content: '';
    display: block;
    width: 140%;
    height: 100%;
    left: -20%;
    z-index: 1000;
    transition: all ease-in-out 0.5s;
    background-repeat: no-repeat;
}
.first-use-button::before {
    display: none;
    top: -75%;
    background-image: radial-gradient(circle, #00b4f2 20%, transparent 20%),
        radial-gradient(circle, transparent 12%, #f96216 20%, transparent 30%),
        radial-gradient(circle, #f96216 20%, transparent 20%), radial-gradient(circle, #00b4f2 20%, transparent 20%),
        radial-gradient(circle, transparent 10%, #b200f8 15%, transparent 20%),
        radial-gradient(circle, #b200f8 20%, transparent 20%), radial-gradient(circle, #b200f8 20%, transparent 20%),
        radial-gradient(circle, #00b4f2 20%, transparent 20%), radial-gradient(circle, #f96216 20%, transparent 20%);
    background-size: 10% 20%, 20% 40%, 15% 40%, 20% 50%, 18% 40%, 10% 10%, 15% 30%, 10% 50%, 18% 18%;
}
.first-use-button::after {
    display: none;
    bottom: -75%;
    background-image: radial-gradient(circle, #00b4f2 20%, transparent 20%),
        radial-gradient(circle, #f96216 20%, transparent 20%),
        radial-gradient(circle, transparent 8%, #00b4f2 15%, transparent 20%),
        radial-gradient(circle, #b200f8 20%, transparent 20%), radial-gradient(circle, #00b4f2 20%, transparent 20%),
        radial-gradient(circle, #00b4f2 20%, transparent 20%), radial-gradient(circle, #b200f8 20%, transparent 20%);
    background-size: 15% 30%, 20% 50%, 18% 40%, 20% 50%, 15% 30%, 10% 20%, 20% 40%;
}
.first-use-button.animate::before {
    display: block;
    animation: topBubbles ease-in-out 0.75s forwards;
}
.first-use-button.animate::after {
    display: block;
    animation: bottomBubbles ease-in-out 0.75s forwards;
}

@keyframes topBubbles {
    0% {
        background-position: 5% 120%, 10% 120%, 10% 120%, 15% 120%, 25% 120%, 25% 100%, 40% 100%, 55% 100%, 70% 100%;
    }
    50% {
        background-position: 0% 100%, 0% 50%, 10% 80%, 20% 0%, 30% 30%, 22% 50%, 50% 50%, 65% 20%, 90% 30%;
    }
    100% {
        background-position: 0% 70%, 0% 100%, 10% 100%, 20% -50%, 30% 70%, 22% 40%, 50% 80%, 65% 10%, 90% 70%;
        background-size: 0% 0%, 0% 0%, 0% 0%, 0% 0%, 0% 0%, 0% 0%;
    }
}
@keyframes bottomBubbles {
    0% {
        background-position: 10% -40%, 30% 10%, 55% -10%, 70% -10%, 85% -10%, 70% -10%, 70% 0%;
    }
    50% {
        background-position: 0% 80%, 20% 80%, 5% 60%, 60% 100%, 75% 70%, 95% 60%, 105% 0%;
    }
    100% {
        background-position: -40% 140%, 20% 140%, 45% 140%, 60% 140%, 75% 130%, 95% 70%, 120% 100%;
        background-size: 0% 0%, 0% 0%, 0% 0%, 0% 0%, 0% 0%, 0% 0%;
    }
}
