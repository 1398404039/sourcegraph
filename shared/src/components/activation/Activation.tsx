import H from 'history'
import { BehaviorSubject, Observable, Subject } from 'rxjs'
import { first, pairwise } from 'rxjs/operators'

export interface ActivationProps {
    activation?: ActivationStatus
}

export interface ActivationStep {
    id: string
    title: string
    detail: string
    action: (h: H.History) => void
}

/**
 * Tracks the activation status of the current user.
 */
export class ActivationStatus {
    public readonly steps: ActivationStep[]
    private fetchCompleted: () => Observable<{ [key: string]: boolean }>

    /**
     * The completion status. Null indicates that the completion status has never been
     * fetched from the server. After initialization, this value will always be
     * non-null.
     */
    public readonly completed: BehaviorSubject<{ [key: string]: boolean } | null>

    /**
     * Promise that resolves after initialization (when the current value of completed is not null).
     */
    private initialized: Promise<void>

    private updateCompleted_: Subject<{ [key: string]: boolean }>

    /**
     * Accepts the activation steps to track and a function that fetches the completed
     * activation steps (in the form of a function that returns an Observable).
     */
    constructor(steps: ActivationStep[], fetchCompleted: () => Observable<{ [key: string]: boolean }>) {
        this.steps = steps
        this.fetchCompleted = fetchCompleted
        this.completed = new BehaviorSubject<{ [key: string]: boolean } | null>(null)
        this.updateCompleted_ = new Subject<{ [key: string]: boolean }>()
        this.initialized = new Promise<void>(resolve => {
            this.completed
                .pipe(
                    pairwise(),
                    first()
                )
                .subscribe(() => resolve())
        })
        this.refetch() // trigger the first fetch
    }

    /**
     * Returns an Observable that fires whenever the client triggers an update in activation status.
     */
    public get updateCompleted(): Observable<{ [key: string]: boolean }> {
        return this.updateCompleted_
    }

    public refetch(): void {
        this.fetchCompleted()
            .pipe(first()) // subscription will get auto-cleaned up
            .subscribe(res => {
                const nextCompleted: { [key: string]: boolean } = {}
                for (const step of this.steps) {
                    nextCompleted[step.id] = res[step.id] || false
                }
                this.completed.next(nextCompleted)
            })
    }

    /**
     * Updates the current completion status with the partial completition status `u`.
     * If the completion status has not yet been fetched (`this.initialized` is unresolved),
     * the update will only be applied after the completion status has been fetched.
     */
    public update(u: { [key: string]: boolean }): void {
        this.initialized.then(() => {
            this.updateCompleted_.next(u)

            if (!this.completed.value) {
                return
            }
            const newVal: { [key: string]: boolean } = {}
            Object.assign(newVal, this.completed.value)
            for (const step of this.steps) {
                if (u[step.id] !== undefined) {
                    newVal[step.id] = u[step.id]
                }
            }
            this.completed.next(newVal)
        })
    }
}

/**
 * Returns the percent of activation checklist items completed.
 */
export const percentageDone = (info?: { [key: string]: boolean }): number => {
    if (!info) {
        return 0
    }
    const vals = Object.values(info)
    return (100 * vals.filter(e => e).length) / vals.length
}
