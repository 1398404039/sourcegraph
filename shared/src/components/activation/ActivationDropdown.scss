.activation-dropdown {
    width: 21rem;
    z-index: 999;

    &__checklist {
        .list-group-item {
            border: 0;
        }

        .activation-dropdown-header {
            border: 0;
        }
    }

    [data-reach-accordion-button] {
        color: var(--link-color);
        .icon-down {
            display: none;
        }
        .icon-right {
            display: inline-block;
        }
    }

    [data-reach-accordion-button][aria-expanded='true'] {
        background-color: $dropdown-link-hover-bg;
        color: var(--link-color);

        .icon-right {
            display: none;
        }
        .icon-down {
            display: inline-block;
        }
    }

    [data-reach-accordion-panel][data-state='open'] {
        background-color: $dropdown-link-hover-bg;
    }
}

.activation-dropdown-button {
    display: flex;
    border: none;
    color: var(--link-color);

    &--hidden {
        display: none;
    }

    &__progress-bar-container {
        display: inline-block;
        width: 1rem;
        margin-top: -0.125rem;
        margin-left: 0.375rem;
    }
    &__circular-progress-bar {
        // stylelint-disable-next-line selector-class-pattern
        .CircularProgressbar-trail {
            stroke: $dropdown-border-color;
        }
        // stylelint-disable-next-line selector-class-pattern
        .CircularProgressbar-path {
            stroke: $oc-green-4;
        }
    }
    &__loader {
        padding-left: 0.8375rem;
    }

    &__confetti {
        z-index: 1000;
    }
}

.activation-dropdown-header {
    white-space: normal;
    border-bottom: 1px solid $dropdown-border-color;
    padding: 0.25rem 0.5rem 0 0.5rem;

    &:hover {
        text-decoration: none;
    }
}
