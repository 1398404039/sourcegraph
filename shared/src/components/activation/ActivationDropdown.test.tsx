import * as H from 'history'
import React from 'react'
import renderer from 'react-test-renderer'
import { noop } from 'rxjs'
import { Activation } from './Activation'
import { ActivationDropdown } from './ActivationDropdown'
import ReactDOM from 'react-dom'

jest.mock('mdi-react/CheckboxBlankCircleIcon', () => 'CheckboxBlankCircleIcon')
jest.mock('mdi-react/CheckIcon', () => 'CheckIcon')

describe('ActivationDropdown', () => {
    const baseActivation: Activation = {
        steps: [
            {
                id: 'ConnectedCodeHost',
                title: 'title1',
                detail: 'detail1',
                onClick: noop,
            },
            {
                id: 'EnabledRepository',
                title: 'title2',
                detail: 'detail2',
            },
        ],
        refetch: noop,
        update: noop,
    }
    const history = H.createMemoryHistory({ keyLength: 0 })
    test('render loading', () => {
        const component = renderer.create(<ActivationDropdown history={history} activation={baseActivation} />, {
            createNodeMock: reactElement => {
                console.log(reactElement)
                return document.createElement(reactElement.type as string)
            },
        })
        expect(component.toJSON()).toMatchSnapshot()
    })
    // test('render 0/2 completed', () => {
    //     const component = renderer.create(
    //         <ActivationDropdown history={history} activation={{ ...baseActivation, completed: {} }} />,
    //         {
    //             createNodeMock: () => ({ ownerDocument: document.implementation.createHTMLDocument() }),
    //         }
    //     )
    //     expect(component.toJSON()).toMatchSnapshot()
    // })
    // test('render 1/2 completed', () => {
    //     const component = renderer.create(
    //         <ActivationDropdown
    //             history={history}
    //             activation={{ ...baseActivation, completed: { ConnectedCodeHost: true } }}
    //         />,
    //         {
    //             createNodeMock: () => ({ ownerDocument: document.implementation.createHTMLDocument() }),
    //         }
    //     )
    //     expect(component.toJSON()).toMatchSnapshot()
    // })
})
