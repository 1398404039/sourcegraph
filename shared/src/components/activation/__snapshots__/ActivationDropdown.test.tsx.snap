// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivationDropdown render 0/2 completed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-controls="menu--5"
        aria-haspopup="true"
        class="activation-dropdown-button activation-dropdown-button__animated-button bg-transparent align-items-center e2e-activation-nav-item-toggle activation-dropdown-button--hidden"
        data-reach-menu-button=""
        id="menu-button--menu--5"
        type="button"
      >
        <div
          class="activation-dropdown-button__confetti"
        >
          <div
            style="position: relative;"
          />
        </div>
        Get started
        <div
          class="activation-dropdown-button__confetti"
        >
          <div
            style="position: relative;"
          />
        </div>
        <span
          class="activation-dropdown-button__progress-bar-container"
        >
          <svg
            class="CircularProgressbar activation-dropdown-button__circular-progress-bar"
            data-test-id="CircularProgressbar"
            viewBox="0 0 100 100"
          >
            <path
              class="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
              fill-opacity="0"
              stroke-width="12"
              style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
            />
            <path
              class="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
              fill-opacity="0"
              stroke-width="12"
              style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: NaNpx;"
            />
          </svg>
        </span>
      </button>
    </div>
    <reach-portal>
      <div
        data-reach-menu=""
        data-reach-menu-popover=""
        data-reach-popover=""
        hidden=""
        style="position: absolute; left: 0px; top: 0px;"
      />
    </reach-portal>
    <reach-portal>
      <div
        data-reach-menu=""
        data-reach-menu-popover=""
        data-reach-popover=""
        hidden=""
        style="position: absolute; left: 0px; top: 0px;"
      >
        <div
          aria-labelledby="menu-button--menu--5"
          class="activation-dropdown"
          data-reach-menu-items=""
          data-reach-menu-list=""
          id="menu--5"
          role="menu"
          tabindex="-1"
        >
          <div
            class="dropdown-item-text activation-dropdown-header"
          >
            <h3
              class="mb-2"
            >
              Welcome to Sourcegraph!
            </h3>
            <p
              class="mb-2"
            >
              Complete the steps below to finish onboarding!
            </p>
          </div>
          <div
            class="activation-dropdown__list"
            data-reach-accordion=""
          >
            <div>
              <div
                data-reach-accordion-item=""
                data-state="collapsed"
              >
                <button
                  aria-controls="panel--7--0"
                  aria-expanded="false"
                  class="activation-dropdown-item dropdown-item"
                  data-reach-accordion-button=""
                  id="button--7--0"
                >
                  <div
                    class="activation-checklist-item d-flex justify-content-between activation-checklist-item--dropdown"
                  >
                    <div>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-down"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-right"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist__title"
                      >
                        title1
                      </span>
                    </div>
                    <div>
                      <svg
                        class="mdi-icon icon-inline text-muted"
                        fill="currentColor"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-labelledby="button--7--0"
                    class="px-2"
                    data-reach-accordion-panel=""
                    data-state="collapsed"
                    hidden=""
                    id="panel--7--0"
                    role="region"
                    tabindex="-1"
                  >
                    <div
                      class="activation-dropdown-item__detail pb-1"
                    >
                      detail1
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div>
              <div
                data-reach-accordion-item=""
                data-state="collapsed"
              >
                <button
                  aria-controls="panel--7--1"
                  aria-expanded="false"
                  class="activation-dropdown-item dropdown-item"
                  data-reach-accordion-button=""
                  id="button--7--1"
                >
                  <div
                    class="activation-checklist-item d-flex justify-content-between activation-checklist-item--dropdown"
                  >
                    <div>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-down"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-right"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist__title"
                      >
                        title2
                      </span>
                    </div>
                    <div>
                      <svg
                        class="mdi-icon icon-inline text-muted"
                        fill="currentColor"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-labelledby="button--7--1"
                    class="px-2"
                    data-reach-accordion-panel=""
                    data-state="collapsed"
                    hidden=""
                    id="panel--7--1"
                    role="region"
                    tabindex="-1"
                  >
                    <div
                      class="activation-dropdown-item__detail pb-1"
                    >
                      detail2
                    </div>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </reach-portal>
  </body>,
  "container": <div>
    <button
      aria-controls="menu--5"
      aria-haspopup="true"
      class="activation-dropdown-button activation-dropdown-button__animated-button bg-transparent align-items-center e2e-activation-nav-item-toggle activation-dropdown-button--hidden"
      data-reach-menu-button=""
      id="menu-button--menu--5"
      type="button"
    >
      <div
        class="activation-dropdown-button__confetti"
      >
        <div
          style="position: relative;"
        />
      </div>
      Get started
      <div
        class="activation-dropdown-button__confetti"
      >
        <div
          style="position: relative;"
        />
      </div>
      <span
        class="activation-dropdown-button__progress-bar-container"
      >
        <svg
          class="CircularProgressbar activation-dropdown-button__circular-progress-bar"
          data-test-id="CircularProgressbar"
          viewBox="0 0 100 100"
        >
          <path
            class="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
            fill-opacity="0"
            stroke-width="12"
            style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
          />
          <path
            class="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
            fill-opacity="0"
            stroke-width="12"
            style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: NaNpx;"
          />
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ActivationDropdown render 1/2 completed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-controls="menu--9"
        aria-haspopup="true"
        class="activation-dropdown-button activation-dropdown-button__animated-button bg-transparent align-items-center e2e-activation-nav-item-toggle activation-dropdown-button--hidden"
        data-reach-menu-button=""
        id="menu-button--menu--9"
        type="button"
      >
        <div
          class="activation-dropdown-button__confetti"
        >
          <div
            style="position: relative;"
          />
        </div>
        Get started
        <div
          class="activation-dropdown-button__confetti"
        >
          <div
            style="position: relative;"
          />
        </div>
        <span
          class="activation-dropdown-button__progress-bar-container"
        >
          <svg
            class="CircularProgressbar activation-dropdown-button__circular-progress-bar"
            data-test-id="CircularProgressbar"
            viewBox="0 0 100 100"
          >
            <path
              class="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
              fill-opacity="0"
              stroke-width="12"
              style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
            />
            <path
              class="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
              fill-opacity="0"
              stroke-width="12"
              style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
            />
          </svg>
        </span>
      </button>
    </div>
    <reach-portal>
      <div
        data-reach-menu=""
        data-reach-menu-popover=""
        data-reach-popover=""
        hidden=""
        style="position: absolute; left: 0px; top: 0px;"
      />
    </reach-portal>
    <reach-portal>
      <div
        data-reach-menu=""
        data-reach-menu-popover=""
        data-reach-popover=""
        hidden=""
        style="position: absolute; left: 0px; top: 0px;"
      >
        <div
          aria-labelledby="menu-button--menu--9"
          class="activation-dropdown"
          data-reach-menu-items=""
          data-reach-menu-list=""
          id="menu--9"
          role="menu"
          tabindex="-1"
        >
          <div
            class="dropdown-item-text activation-dropdown-header"
          >
            <h3
              class="mb-2"
            >
              Almost there!
            </h3>
            <p
              class="mb-2"
            >
              Complete the steps below to finish onboarding!
            </p>
          </div>
          <div
            class="activation-dropdown__list"
            data-reach-accordion=""
          >
            <div>
              <div
                data-reach-accordion-item=""
                data-state="collapsed"
              >
                <button
                  aria-controls="panel--11--0"
                  aria-expanded="false"
                  class="activation-dropdown-item dropdown-item"
                  data-reach-accordion-button=""
                  id="button--11--0"
                >
                  <div
                    class="activation-checklist-item d-flex justify-content-between activation-checklist-item--dropdown"
                  >
                    <div>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-down"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-right"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist__title"
                      >
                        title1
                      </span>
                    </div>
                    <div>
                      <svg
                        class="mdi-icon icon-inline text-success"
                        fill="currentColor"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-labelledby="button--11--0"
                    class="px-2"
                    data-reach-accordion-panel=""
                    data-state="collapsed"
                    hidden=""
                    id="panel--11--0"
                    role="region"
                    tabindex="-1"
                  >
                    <div
                      class="activation-dropdown-item__detail pb-1"
                    >
                      detail1
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div>
              <div
                data-reach-accordion-item=""
                data-state="collapsed"
              >
                <button
                  aria-controls="panel--11--1"
                  aria-expanded="false"
                  class="activation-dropdown-item dropdown-item"
                  data-reach-accordion-button=""
                  id="button--11--1"
                >
                  <div
                    class="activation-checklist-item d-flex justify-content-between activation-checklist-item--dropdown"
                  >
                    <div>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-down"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist-item__icon icon-inline icon-right"
                      >
                        <svg
                          class="mdi-icon "
                          fill="currentColor"
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="activation-checklist__title"
                      >
                        title2
                      </span>
                    </div>
                    <div>
                      <svg
                        class="mdi-icon icon-inline text-muted"
                        fill="currentColor"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-labelledby="button--11--1"
                    class="px-2"
                    data-reach-accordion-panel=""
                    data-state="collapsed"
                    hidden=""
                    id="panel--11--1"
                    role="region"
                    tabindex="-1"
                  >
                    <div
                      class="activation-dropdown-item__detail pb-1"
                    >
                      detail2
                    </div>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </reach-portal>
  </body>,
  "container": <div>
    <button
      aria-controls="menu--9"
      aria-haspopup="true"
      class="activation-dropdown-button activation-dropdown-button__animated-button bg-transparent align-items-center e2e-activation-nav-item-toggle activation-dropdown-button--hidden"
      data-reach-menu-button=""
      id="menu-button--menu--9"
      type="button"
    >
      <div
        class="activation-dropdown-button__confetti"
      >
        <div
          style="position: relative;"
        />
      </div>
      Get started
      <div
        class="activation-dropdown-button__confetti"
      >
        <div
          style="position: relative;"
        />
      </div>
      <span
        class="activation-dropdown-button__progress-bar-container"
      >
        <svg
          class="CircularProgressbar activation-dropdown-button__circular-progress-bar"
          data-test-id="CircularProgressbar"
          viewBox="0 0 100 100"
        >
          <path
            class="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
            fill-opacity="0"
            stroke-width="12"
            style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
          />
          <path
            class="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
            fill-opacity="0"
            stroke-width="12"
            style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
          />
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ActivationDropdown render loading 1`] = `
<div>
  <button
    aria-controls="menu--1"
    aria-haspopup="true"
    class="activation-dropdown-button activation-dropdown-button__animated-button bg-transparent align-items-center e2e-activation-nav-item-toggle activation-dropdown-button--hidden"
    data-reach-menu-button=""
    id="menu-button--menu--1"
    type="button"
  >
    <div
      class="activation-dropdown-button__confetti"
    >
      <div
        style="position: relative;"
      />
    </div>
    Get started
    <div
      class="activation-dropdown-button__confetti"
    >
      <div
        style="position: relative;"
      />
    </div>
    <span
      class="activation-dropdown-button__progress-bar-container"
    >
      <svg
        class="CircularProgressbar activation-dropdown-button__circular-progress-bar"
        data-test-id="CircularProgressbar"
        viewBox="0 0 100 100"
      >
        <path
          class="CircularProgressbar-trail"
          d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
          fill-opacity="0"
          stroke-width="12"
          style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
        />
        <path
          class="CircularProgressbar-path"
          d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
          fill-opacity="0"
          stroke-width="12"
          style="stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 276.46015351590177px;"
        />
      </svg>
    </span>
  </button>
</div>
`;
